python-package/lightgbm/plotting.py

```diff
@@ -369,8 +402,15 @@ def plot_metric(booster, metric=None, dataset_names=None,
     return ax
 
 
-def _to_graphviz(tree_info, show_info, feature_names, precision=3,
-                 orientation='horizontal', constraints=None, **kwargs):
+def _to_graphviz(
+    tree_info, 
+    show_info: Optional[List[str]], 
+    feature_names, 
+    precision: Optional[int] = 3,
+    orientation: Optional[str] = 'horizontal', 
+    constraints = None, 
```

Comment:
```suggestion
    constraints: Optional[List[int]]
```

I know this argument is not documented, but I looked at the source code + LightGBM's tests to work backwards to the shape of it.

https://github.com/WestonKing-Leatham/LightGBM/blob/5ed047233f4787100aa76d82dad1f90abf84e8de/python-package/lightgbm/plotting.py#L575

https://github.com/microsoft/LightGBM/blob/a8fae81fc64a52d1f0b3fae5019870d7609d64c2/tests/python_package_test/test_plotting.py#L134

Then ran the following code to figure it out:

```python
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split

import lightgbm as lgb

X_train, _, y_train, _ = train_test_split(
    *load_breast_cancer(return_X_y=True),
    test_size=0.1,
    random_state=1
)

constraints = [-1, 1] * int(X_train.shape[1] / 2)
gbm = lgb.LGBMClassifier(n_estimators=10, num_leaves=3, silent=True, monotone_constraints=constraints)
gbm.fit(X_train, y_train, verbose=False)

model = gbm.booster_.dump_model()
model.get("monotone_constraints", None)
```